### DISCUSSION TITLE
Replacement for redis-namespace

### DISCUSSION THREAD

**Discussion by jonathan-anyword** [sidekiq/sidekiq#6563] [Q&A]
Hi All,
Since Sidekiq Ent support health-check form version 7.1.2 we need to do migration to this version or higher.
The problem that our Staging env architecture is Strong relays on redis-namespace so we can share one redis for many envs, Each env has its own Sidekiq.

Is any solution to replace is that is not logical DBs? there are only 16 on redis.

Thanks

**Comment by mperham** [comment]
Redis can support as many numbered DBs as you want, you just need to configure it in redis.conf. 16 is the default.

I don’t know of any other solution but I don’t know anything about your staging setup.

**Comment by jonathan-anyword** [comment]
Is Sidekiq support internally on logical clusters separation? today we achieve that with this logic:
```
sidekiq_redis_opt = if Rails.env.staging? # || Rails.env.development?
                      { url: ENV['SIDEKIQ_REDIS_URL'], namespace: ENV.fetch('ENV_LOGIC_NAME') }
                    else
                      { url: ENV['SIDEKIQ_REDIS_URL'] }
                    end
```

**Reply by mperham** [reply]
Did you read the Using Redis wiki page?

**Reply by jonathan-anyword** [reply]
Today the logic is that everyone can add create staging env with any logical name he wants and get small cloud environment ready to use.

The namespace helped us to achieve that with unique string that is the name of the env.

I see now that it is possible to create more DBs with the configuration of redis, I hope the AWS elastic cache supporting it.

One option that I have is to create logic that sit on the common DB on the redis and assign unique number for every env created.
the assign logic is pretty easy but the releasing part need another external logic that will remove the assignment. 

Do you see any easier solution to achieve that?

**Reply by mperham** [reply]
And why can’t the staging environment provision a local Redis instance to use also?

**Reply by jonathan-anyword** [reply]
It is possible but requires infra and devops resources. The elasticache is the production solution and it easy to implement.

I have implemented is logic with code on the startup and it seems to be working.
Again it is not the best solution since it is staging specific but the namespace was staging specific also...
